docs/source/sft_trainer.mdx

```diff
@@ -0,0 +1,154 @@
+
+## Quickstart
+
+If you have a dataset hosted on 🤗 Hub, you can easily fine-tune your SFT model using `SFTTrainer` from TRL. Let us assume your dataset is `imdb` , the text you want to predict is inside `text` field of the dataset and you want to fine-tune `facebook/opt-350m` model.
+
+```python
+from datasets import load_dataset
+from trl import SFTTrainer
+
+dataset = load_dataset("imdb", split="train")
+
+trainer = SFTTrainer(
+    "facebook/opt-350m",
+    train_dataset=dataset,
+    dataset_text_field="text",
+)
+```
+
+Of course you can also construct a model outside of the trainer and pass it to the trainer. For example:
+
+```python
+from transformers import AutoModelForCausalLM
+from datasets import load_dataset
+from trl import SFTTrainer
+
+dataset = load_dataset("imdb", split="train")
+
+model = AutoModelForCausalLM.from_pretrained("facebook/opt-350m")
+
+trainer = SFTTrainer(
+    model,
+    train_dataset=dataset,
+    dataset_text_field="text",
+)
+
+trainer.train()
+```
+
+The above snippets will use the default training arguments from [`transformers.TrainingArguments`](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments) class. If you want to modify that, make sure to create your own `TrainingArguments` object and pass it to the `SFTTrainer` constructor as it is done on the [`supervised_finetuning.py` script](https://github.com/lvwerra/trl/blob/main/examples/stack_llama/scripts/supervised_finetuning.py) on the stack-llama example.
+
+
+## Advanced usage
+
+### Packing dataset (`ConstantLengthDataset`)
+
+`SFTTrainer` supports the integration of packed dataset. This is done with the `ConstantLengthDataset` utility clss that returns constant length chunks of tokens from stream of text files. To enable the usage of this dataset class, simply pass `packing=True` to the `SFTTrainer` constructor.
+
+```python
+...
+
+trainer = SFTTrainer(
+    "facebook/opt-350m",
+    train_dataset=dataset,
+    dataset_text_field="text",
+    packing=True
+)
+
+trainer.train()
+```
+
+If your dataset has several fields that you want to combine, for example if the dataset has `question` and `answer` fields and you want to combine them, you can pass a formatting function to the trainer that will take care of that. For example:
+
+```python
+def formatting_func(example):
+    text = f"### Question: {example['question']}\n ### Answer: {example['answer']}"
+    return text
+
+trainer = SFTTrainer(
+    "facebook/opt-350m",
+    train_dataset=dataset,
+    dataset_text_field="text",
+    packing=True,
+    formatting_func=formatting_func
+)
+
+trainer.train()
+```
+You can also customize the `ConstantLengthDataset` much more by directly passing the arguments to the `SFTTrainer` constructor. Please refer to that class' signature for more information.
```

Comment:
Should these references to the classes include doc links?

Reply:
The references should be now added!